src/ngraph/runtime/intelgpu/intelgpu_backend.hpp

```diff
@@ -68,12 +86,19 @@ class ngraph::runtime::intelgpu::IntelGPUBackend : public runtime::Backend
         bool m_performance_counters_enabled = false;
         double m_compilation_time = 0.0;
         double m_consumed_memory = 0.0;
-    };
-
-    std::map<std::shared_ptr<Function>, FunctionInstance> ocl_networks;
-    std::shared_ptr<cldnn::engine> ocl_engine;
+        std::shared_ptr<Function> m_function;
+    } m_function_instance;
 
+    bool m_profile_enable = false;
```

Comment:
Variables are duplicated in `Backend` and `Executable`. But this is not a problem for this PR. I will take care about this later.
Please proceed with main idea of the PR.

Reply:
Thank you. Yes, it is not an ideal solution but I believe you can see the path to get to a better solution. We can do that as individual, backend focused PRs.